### PR TITLE
Add php codesniffer pre-commit hook

### PR DESCRIPTION
**Pull request by Wharenn**

### CODE REVIEW

File: php/phpcs/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+## Features
+
+ * Check Coding style and forbid the commit if violations are found
+ * Configuration file for Coding Standard, Path to PHPCS, Ignore List
+ * Shows output in a 'less' pipe following the smart git principles
+
+
+## Usage
+
+Installation intructions:
```
Comment (eko) [inline php/phpcs/README.md]: There is a "s" missing here: "instructions" ;)

File: php/phpcs/config [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+PHPCS_BIN=/usr/bin/phpcs
+
+# the coding standard, you can also specify a path to your own standard here
+# e. g. /path/to/my/standard/dir/
+PHPCS_CODING_STANDARD=PEAR
+
+# comma-separated list of file patterns being ignored
+PHPCS_IGNORE=
+
+# comma-seperated list of sniffs from the standard that should be used
```
Comment (eko) [inline php/phpcs/config]: There is a typo here: `seperated` -> `separated`

File: php/phpcs/config [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# use `phpcs --standard=PSR1 -e` to list sniffs for your standard
+PHPCS_SNIFFS=Generic.Files.ByteOrderMark,Generic.PHP.DisallowShortOpenTag
+
+# egrep compatible pattern of  files to be checked
```
Comment (eko) [inline php/phpcs/config]: There is a double space here between `of` and `files`

**Comment by eko** [comment]
This is cool :+1:

**Comment by Wharenn** [comment]
Thanks @eko

**Comment by plouc** [comment]
Can you please update the main README to provide a link to the php part ?

File: php/phpcs/pre-commit [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    exit 1
+fi
+
+# stolen from template file
+if git rev-parse --verify HEAD
+then
+    against=HEAD
+else
+    # Initial commit: diff against an empty tree object
+    against=4b825dc642cb6eb9a060e54bf8d69288fbee4904
```
Comment (amine2z) [inline php/phpcs/pre-commit:32]: its normal that this sha1 is here?

Comment (Wharenn) [inline php/phpcs/pre-commit:32]: This is the empty tree sha1 given by git. More info [here](http://stackoverflow.com/questions/9765453/gits-semi-secret-empty-tree).

**Comment by Wharenn** [comment]
Done @plouc :)

**Comment by plouc** [comment]
:+1:

**Pull request merged**